context/AuthContextProvider.tsx

```diff
@@ -32,16 +37,29 @@ export const AuthContextProvider = ({
         return;
       }
 
+      return setIsSignedIn(true);
+    };
+    checkSession();
+  }, []);
+
+  useMemo(() => {
+    const getUser = async () => {
+      if (!isSignedIn) {
+        return;
+      }
+
       try {
-        await account.getSession('current');
-        setIsSignedIn(true);
+        const userData = await account.get();
+        updateUserId(userData.$id);
+        updateUserEmail(userData.email);
       } catch (error) {
-        console.error('Session validation error:', error);
+        console.error('Error getting user data:', error);
```

Comment:
Add a `throw New Error()` as well. We want the app to crash so the user knows something critical happend.